### PR TITLE
584 image field on dataset metadata v2

### PR DESCRIPTION
**Pull request by travissarles**

### CODE REVIEW

File: knowledge-graph/src/main/scala/ch/datascience/knowledgegraph/datasets/rest/BaseDetailsFinder.scala [hunk 1]
```diff
@@ -56,18 +58,18 @@ private class BaseDetailsFinder(
       renku  -> "renku",
       schema -> "schema"
     ),
-    s"""|SELECT DISTINCT ?identifier ?name ?alternateName ?url ?topmostSameAs ?maybeDerivedFrom ?initialVersion ?description ?publishedDate
+    s"""|SELECT DISTINCT ?identifier ?name ?alternateName ?url ?topmostSameAs ?maybeDerivedFrom ?initialVersion ?description ?publishedDate ?imageIdentifier
```
Comment (travissarles) [inline knowledge-graph/src/main/scala/ch/datascience/knowledgegraph/datasets/rest/BaseDetailsFinder.scala:61]: remove ?imageIdentifier

File: knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/DatasetEndpointSpec.scala [hunk 2]
```diff
@@ -234,4 +237,9 @@ class DatasetEndpointSpec extends AnyWordSpec with MockFactory with ScalaCheckPr
     for {
       initial <- cursor.downField("initial").as[InitialVersion]
     } yield DatasetVersions(initial)
+
+//  private implicit lazy val imageUrlDecoder: Decoder[List[ImageUrl]] = cursor =>
```
Comment (travissarles) [inline knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/DatasetEndpointSpec.scala:241]: get rid of this

File: knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/IODatasetFinderSpec.scala [hunk 3]
```diff
@@ -49,12 +49,16 @@ class IODatasetFinderSpec extends AnyWordSpec with InMemoryRdfStore with ScalaCh
...
           val dataset = nonModifiedDatasets(projects = project.copy(created = addedToProject).toGenerator).generateOne
 
+          val dataSetJsonLD = dataset.toJsonLD(noSameAs = true)() //this has no images
           loadToStore(
-            dataset.toJsonLD(noSameAs = true)(),
+            dataSetJsonLD,
             randomDataSetCommit
           )
+          val results = datasetFinder
+            .findDataset(dataset.id)
```
Comment (travissarles) [inline knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/IODatasetFinderSpec.scala:58]: inline

File: knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/IODatasetFinderSpec.scala [hunk 4]
```diff
@@ -49,12 +49,16 @@ class IODatasetFinderSpec extends AnyWordSpec with InMemoryRdfStore with ScalaCh
...
+            .unsafeRunSync()
 
-          datasetFinder.findDataset(dataset.id).unsafeRunSync() shouldBe Some(
+          results shouldBe Some( // the expectation is correct but the finding is not
```
Comment (travissarles) [inline knowledge-graph/src/test/scala/ch/datascience/knowledgegraph/datasets/rest/IODatasetFinderSpec.scala:61]: remove comment

**Review by jachro — APPROVED** [review]

**Pull request merged**